src/ObjCRuntime/PlatformAvailability2.cs

```diff
@@ -92,6 +92,76 @@ internal AvailabilityBaseAttribute ()
 		public override string ToString ()
 		{
 			var builder = new StringBuilder ();
+#if NET
+			switch (AvailabilityKind) {
+			case AvailabilityKind.Introduced:
+				builder.Append ("[SupportedOSPlatform (\"");
+				break;
+			case AvailabilityKind.Obsoleted:
+				switch (Platform) {
+				case PlatformName.iOS:
+					builder.AppendLine ("#if __IOS__");
+					break;
+				case PlatformName.TvOS:
+					builder.AppendLine ("#if __TVOS__");
+					break;
+				case PlatformName.WatchOS:
+					builder.AppendLine ("#if __WATCHOS__");
+					break;
+				case PlatformName.MacOSX:
+					builder.AppendLine ("#if __MACOS__");
+					break;
+				case PlatformName.MacCatalyst:
+					builder.AppendLine ("#if __MACCATALYST__");
+					break;
+				}
+				builder.Append ("[Obsolete (\"Starting with ");
+				break;
+			case AvailabilityKind.Deprecated:
+			case AvailabilityKind.Unavailable:
+				builder.Append ("[UnsupportedOSPlatform (\"");
+				break;
+			}
+
+			switch (Platform) {
+			case PlatformName.iOS:
+				builder.Append ("ios");
+				break;
+			case PlatformName.TvOS:
+				builder.Append ("tvos");
+				break;
+			case PlatformName.WatchOS:
+				builder.Append ("watchos");
+				break;
+			case PlatformName.MacOSX:
+				builder.Append ("macos"); // no 'x'
+				break;
+			case PlatformName.MacCatalyst:
+				builder.Append ("maccatalyst");
+				break;
+			}
+
+			if (Version != null) {
+				builder.Append (Version.Major).Append ('.').Append (Version.Minor);
+				if (Version.Build >= 0)
+					builder.Append ('.').Append (Version.Build);
+			}
+
```

Comment:
This is minor, but why can't you do:
```
builder.Append(Version.ToString (Version.Build >= 0 ? 3 : 2);
```

Reply:
Good catch! I copied the existing (non `NET` code) and did not think of that overload.